airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py

```diff
@@ -17,6 +17,12 @@ class QuotingBehavior(Enum):
     QUOTE_NONE = "Quote None"
 
 
+class InferenceType(Enum):
+    NONE = "None"
+    PRIMITIVE_TYPES_ONLY = "Primitive Types Only"
+    PRIMITIVE_AND_COMPLEX_TYPES = "Primitive and Complex Types"
```

Comment:
That's a misunderstanding on my part, but I think it makes sense to leave fields as strings rather than infer the data. I agree we should leave the Transform step out of the source connector when possible.

I'm not sure how I feel about hiding the feature because it makes the UX weird: A user will see different behavior between existing sources and new ones, but they won't be able to tell why they're different because the field is hidden.

Reply:
I think the answer is no. I tried adding a literal and a string with const=True to the s3 spec and they are not rendered in the UI

Reply:
Alright, I'm on board with hiding the option. We should just make sure to document the unexpected behavior.

Like you said, it'll be easy to make the option visible if we find out there is a real need for it.

Reply:
@maxi297 we can hide the field from the UI by setting airbyte_hidden to True ([here's an example on the FB connector](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py#L217))

Removing `PRIMITIVE_AND_COMPLEX_TYPES` and its related code path makes sense to me if it's not useful

Reply:
The example you shared is for hiding a whole field from a spec, not a specific value from field that has enum values. Am I missing something?

Reply:
I think that's what we should do
1. Hide the field from the spec
2. Set the value in the S3 config adapter

Reply:
And is this a concern that `NONE, `PRIMITIVE_TYPES_ONLY` and `PRIMITIVE_AND_COMPLEX_TYPES` are available as CDK Python code?

Reply:
I think we can get rid of `PRIMITIVE_AND_COMPLEX_TYPES`, but we need `PRIMITIVE_TYPES_ONLY` to be available since we'll use it for S3

Reply:
Removed!